Dockerfile-bot

```diff
@@ -1,6 +1,23 @@
 FROM docker.io/node:alpine
+
+RUN apk -U upgrade --no-cache
+
 WORKDIR /bot
+
 COPY package.json .
-RUN yarn install
-COPY . .
-CMD yarn bot
+COPY yarn.lock .
+COPY babel.config.json .
+COPY tsconfig.json .
+COPY src/ ./src
```

Comment:
Isn't it easier and less error prone to copy all and filter out in the `.dockerignore` file ?

Reply:
I wouldn't say it's easier doing that, it's just _another_ way to do it. 

Personally, I prefer this approach as it's explicit and IMO very obvious what I'm coping over without having to cross check 2 files. I also suspect it would be quite unlikely for us to want to copy over additional files in the future that aren't in the `src` folder, this somewhat forces us to keep organised

Reply:
But that's exactly what we were doing before? You were copying literally everything and relying on the '.dockerignore' file...

I don't think going back and forth on something so trivial is productive, given our time zones - I'm in Australia.

If you truely feel like copying everything like before was better than being explicit then please modify this PR, otherwise it's going to take us months to get anything done 😄

Reply:
Any updates here @chevdor ? I'm hoping to get this through ASAP so that we can start improving other things and adding a test suite

Reply:
@chevdor @AndreiEres